apps/dashboard/app/models/batch_connect/settings.rb

```diff
@@ -0,0 +1,108 @@
+module BatchConnect
+  # An Interactive Application saved parameters from the form.
+  class Settings
+    include ActiveModel::Model
+
+    class << self
+      def find(token, name)
+        settings_values = BatchConnectSettingsReader.new.app_templates(token).fetch(name.to_sym, {})
```

Comment:
I don't think we need the class ` BatchConnectSettingsReader` - I can kinda tell by the way you use it. You never actually keep the object around, all you do is use `.new.other_method`, so it's sort of like why not just put `other_method` as a class method on Settings?

Reply:
I could not find a way of doing that. The problem is with `UserSettingStore`, I cannot use the module with `include` and use it in a static context. I tried `extend`, then was having problems with instance methods.

This is the only compromise that I found to use static methods with `UserSettingStore`.

Reply:
As well, there was a caching issue when `UserSettingStore` is used in a static context, as the data was not refreshed because the @user_settings field became a class field.

Reply:
Then why not just keep the APIs in `UserSettingStore`? Looking at `ActiveModel` - I can see for sure that modules are composable if we do want to separate out functionality.

Reply:
Something like this to remove `BatchConnectSettingsReader`
```ruby
module UserBatchConnectSettings
  include UserSettingStore

  # specific methods here
  def find
  end

  def save
  end
  
  def delete
  end
end
```
And then make the `Settings` class just hold the data? and all operations done in the controller?

Reply:
🤦‍♂️ I see - we can't compose them because they'd all depend on whatever actually reads & writes that file.

> Something like this to remove BatchConnectSettingsReader

Yea I think so, I mean the way you're using the class right now with `.new.method_i_actually_want` really looks like it should be a module.

> And then make the Settings class just hold the data? and all operations done in the controller?

Sure to the first, I don't quite know what you mean about the second.

I think the only thing I know for sure is you're using this class just to bundle methods together, and really that's a module because the instantiated object has no value other than as a way to get to it's member methods. So it's a class yes, but really it's just a collection of useful stateless methods which is more like a module.

Reply:
> all the operations done in the controller?

Something like this:

```ruby
class BatchConnect::SettingsController < ApplicationController
  include BatchConnectConcern
  include UserBatchConnectSettings

  def show
    # needed to render the new left menu in all controllers that render the menu.
    # Similar to the existing `set_app_groups`
    @all_saved_settings = new_module_all
    app_token = params[:token]
    settings_name = params[:id]
    settings_values = new_module_find(app_token, settings_name)
    @settings = BatchConnect::Settings.new(app_token, settings_name, settings_values)
    if @settings.outdated?
      flash.now[:alert] = t('dashboard.bc_saved_settings.outdated_message', app_title: @settings.app.title)
    end
  end

  # DELETE /batch_connect/<app_token>/settings/<settings_name>
  def destroy
    app_token = params[:token]
    settings_name = params[:id]
    new_module_delete(app_token, settings_name)
    redirect_to new_batch_connect_session_context_path(token: app_token),
                notice: t('dashboard.bc_saved_settings.deleted_message', settings_name: settings.name)
  end
```

Reply:
No i don't mean that much logic in a controller. I'll hack around on this tomorrow and report back.

Reply:
OK I see what's going on here. Yea I think the easiest way forward is to just keep all these APIs in the `UserSettingStore` module. That's the easiest way forward - yes it'll have a bunch of APIs, but it'll at least let us defer composition refactoring later.

Reply:
OK - will make those changes and see how it looks

Reply:
I made some changes, let me know if this is the right direction